### DISCUSSION TITLE
Jekyll Github Pages 404 Error

### DISCUSSION THREAD

**Discussion by Sam-Robertson** [community/community#135544] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I've been working on my first Jekyll site (https://github.com/Sam-Robertson/Scorched), but whenever I push changes my website stops working.

I'm not sure if I'm doing something wrong in my file structure. There have been several times when it started working again after 30 min or so, but I can't afford to have my website go down every time I push new changes.

**Comment by LiteBrite82** [comment]
Hi @Sam-Robertson 
Can you provide some more context? screenshots? error messages? 

This is what I see: 
<img width="1718" alt="Screenshot 2024-08-12 at 10 16 47 AM" src="https://github.com/user-attachments/assets/c1727107-54bb-44d5-9054-4b575dcc845f">

**Reply by davevad93** [reply]
@Sam-Robertson , I confirm what @LiteBrite82 said, when I go to https://sam-robertson.github.io/Scorched/ I see the following home page:

![00](https://github.com/user-attachments/assets/ab5b252e-57d1-4ddf-9bc3-f80481a30baa)

**Reply by Sam-Robertson** [reply]
@davevad93 Do you have an ideas why I would be getting the errors that are shown in the screenshot from an incognito tab? I've run it locally using bundle exec jekyll serve --baseurl="/Scorched" and it's working great on my computer locally. 

I would think it's a problem with how my asset paths are handled, but I don't understand what's going wrong.

**Reply by davevad93** [reply]
> Your resources are being loaded from `/`, but your site is being built and served under `/Scorched`. That's why you're seeing so many 404s.

See the reply from @dane below.

**Comment by Sam-Robertson** [comment]
Interesting. When I go to https://sam-robertson.github.io/Scorched/ it redirects me to scorchedprovo.com (as it should) which shows the following error.

<img width="1451" alt="Screenshot 2024-08-12 at 11 33 10 AM" src="https://github.com/user-attachments/assets/365b42ca-7024-472a-9d19-ca6454ebf791">

@LiteBrite82 What you are seeing is my landing page, but none of the stylings are coming through.

**Reply by Sam-Robertson** [reply]
@LiteBrite82 I just opened an incognito tab and opened up the site and got the same thing

<img width="1486" alt="Screenshot 2024-08-12 at 11 41 39 AM" src="https://github.com/user-attachments/assets/843a1774-b8f1-4c5f-bdda-178313c1c3b4">

It looks like maybe my browser's cache is throwing things off. I'm still not sure why my website runs perfectly locally, but I'm getting these errors when I publish my page.

**Comment by dane** [comment]
Your resources are being loaded from `/`, but your site is being built and served under `/Scorched`. That's why you're seeing so many 404s.

**Reply by BitlifeModAPK** [reply]
The 404 error on GitHub Pages with Jekyll usually happens because of one of the following reasons:

BaseURL Issue: If you're using a custom domain or your site isn't hosted at the root, you may need to set the baseurl parameter correctly in your _config.yml file. For example:

yaml
Copy code
baseurl: "/your-repo-name"
Make sure to use the correct path where your site is hosted.

Improper File Structure: Double-check that your files are in the correct structure within the repository. Ensure that your index.html or index.md file is in the root directory or in the appropriate folder.

CNAME Configuration: If you're using a custom domain, ensure your CNAME file is properly configured and points to your GitHub Pages domain.

Deployment Delays: Sometimes, GitHub Pages may take a few minutes to update and deploy changes, so it's worth waiting a bit or forcing a redeploy by pushing a new commit.

If you're still facing issues, you might want to explore other hosting solutions or configurations. For example, I’ve found that simplifying the deployment process can be done using tools and tips shared on platforms like [BitLife Apps Pro](https://bitlifeappspro.com/), where we discuss various web development challenges and solutions.

**Comment by dane** [comment]
@Sam-Robertson did you add your domain to the repo Settings > Pages > Custom domain?

**Reply by Sam-Robertson** [reply]
@dane Yeah I did. I'm looking into your other comment about the discrepancy between my resources and by build path. I think that might be a large part of the problem. 

Like I mentioned, the site has worked at some point with the custom domain. I just think something in my asset paths might be throwing the whole thing off.

**Reply by Sam-Robertson** [reply]
@dane I changed the `baseurl: "" ` to `baseurl: "/Scorched" ` and now my website is loading content which is great! I appreciate the help. It looks like I might have to fix all the references to my assets because I am now getting these same errors on the scorchedprovo.com page

<img width="1491" alt="Screenshot 2024-08-12 at 12 15 25 PM" src="https://github.com/user-attachments/assets/760d13e2-acb2-4e52-86f4-4c68c8be0b8d">

**Reply by dane** [reply]
It will be the inverse issue with your custom domain. Assets are probably under `/Scorched` which fixes the github.io domain, but breaks your custom domain.

**Reply by LiteBrite82** [reply]
it looks like you have a routing issue. what happens if you remove `/scorched` from line 29 of _config.yml

**Reply by LiteBrite82** [reply]
Also, can you include a screenshot of your Pages Settings page?

**Reply by LiteBrite82** [reply]
digging deeper: The deployed HTML is pointing to paths that dont exist. /scorched/assets/...... I'm not familiar enough with Jekyll to know how to change that, but that is the issue. To get it to work, it needs to point to /assets/....

Its file path vs URL - these references need to be file paths. 
<img width="1390" alt="Screenshot 2024-08-12 at 11 33 07 AM" src="https://github.com/user-attachments/assets/4528a1b7-b9d5-4e7c-898f-219142a97b09">

**Reply by Sam-Robertson** [reply]
> ave a routing issue. what happens if you remove

@LiteBrite82  I'm testing removing line 29 now

<img width="1436" alt="Screenshot 2024-08-12 at 12 36 13 PM" src="https://github.com/user-attachments/assets/ed2621bd-785d-4048-a3d0-a5bb9226b555">

**Reply by LiteBrite82** [reply]
that looks right

**Reply by LiteBrite82** [reply]
found this article that could be helpful: https://mademistakes.com/mastering-jekyll/site-url-baseurl/
- the comments at the bottom could be useful

**Reply by LiteBrite82** [reply]
_config.yml line 30: change http -> https

**Reply by LiteBrite82** [reply]
screen shot of DNS settings?

**Comment by Sam-Robertson** [comment, marked as answer]
The final answer as far as I can tell was this:

Github Pages requires you to reapply the custom domain every time you push new changes and redeploy your website. I had pushed new updates and hadn't added in the custom domain again. In trying to fix the website, I created more bugs that led me to further misdiagnose the problem.
